cmd/launcher/launcher.go

```diff
@@ -235,25 +176,143 @@ func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) err
+	if err != nil {
+		return fmt.Errorf("writing pid file: %w", err)
+	}
+	return nil
+}
+
+func initRootDirectory(logger log.Logger, opts *launcher.Options) (string, error) {
+	if opts.RootDirectory != "" {
+		return opts.RootDirectory, nil
+	}
+
+	rootDirectory := filepath.Join(os.TempDir(), defaultRootDirectory)
+	if _, err := os.Stat(rootDirectory); os.IsNotExist(err) {
+		if err := os.Mkdir(rootDirectory, fs.DirMode); err != nil {
+			return "", fmt.Errorf("creating temporary root directory: %w", err)
+		}
+	}
+	level.Info(logger).Log(
+		"msg", "using default system root directory",
+		"path", rootDirectory,
+	)
+
+	if err := os.MkdirAll(rootDirectory, 0700); err != nil {
+		return "", fmt.Errorf("creating root directory: %w", err)
+	}
+
+	return rootDirectory, nil
+}
+
+func initHttpClient(opts *launcher.Options) *http.Client {
+	// construct the appropriate http client based on security settings
+	httpClient := http.DefaultClient
+	if opts.InsecureTLS {
+		httpClient = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
+				},
+			},
+		}
+	}
+	return httpClient
+}
+
+func initBoltDb(dir string) (*bbolt.DB, error) {
+	boltOptions := &bbolt.Options{Timeout: time.Duration(30) * time.Second}
+	db, err := bbolt.Open(filepath.Join(dir, "launcher.db"), 0600, boltOptions)
+	if err != nil {
+		return nil, fmt.Errorf("opening launcher db: %w", err)
+	}
+	return db, nil
+}
+
+func createCertificatePool(opts *launcher.Options) (*x509.CertPool, error) {
+	// create the certificate pool
+	var rootPool *x509.CertPool
+	if opts.RootPEM != "" {
+		rootPool = x509.NewCertPool()
+		pemContents, err := ioutil.ReadFile(opts.RootPEM)
+		if err != nil {
+			return nil, fmt.Errorf("reading root certs PEM at path %s: %w", opts.RootPEM, err)
+		}
+		if ok := rootPool.AppendCertsFromPEM(pemContents); !ok {
+			return nil, fmt.Errorf("appending root certs PEM at path %s: %w", opts.RootPEM, err)
+		}
+	}
+	return rootPool, nil
+}
+
+func handleSignals(logger log.Logger, cancel func(), runGroup *run.Group) chan os.Signal {
+	// Create a channel for signals
+	sigChannel := make(chan os.Signal, 1)
+
+	// Add a rungroup to catch things on the sigChannel
+	// Attach a notifier for os.Interrupt
+	runGroup.Add(func() error {
+		signal.Notify(sigChannel, os.Interrupt)
+		select {
+		case <-sigChannel:
+			level.Info(logger).Log("msg", "beginnning shutdown via signal")
+			return nil
+		}
+	}, func(err error) {
+		level.Info(logger).Log("msg", "interrupted", "err", err)
+		level.Debug(logger).Log("msg", "interrupted", "err", err, "stack", fmt.Sprintf("%+v", err))
+		cancel()
+		close(sigChannel)
+	})
+
+	return sigChannel
+}
+
+func createSystrayRunner(logger log.Logger, runGroup *run.Group, opts *launcher.Options) *systrayruntime.SystrayUsersProcessesRunner {
 	if (opts.KolideServerURL == "k2device-preprod.kolide.com" || opts.KolideServerURL == "localhost:3443") && runtime.GOOS == "darwin" {
```

Comment:
For this one specifically, I kinda want the `opts.KolideServerURL == "k2device-preprod.kolide.com" || opts.KolideServerURL == "localhost:3443"` up in the main func. Since I'm thinking about hanging other things off it.

But doing it here does make for a clean return nil, so I'm not sure.

I think it's going to evolve real fast. There are 2 knobs we're going to need. First one is whether it's talking to Kolide, which can pull in `ee`. Second one is whether it's the Kolide account, which is basically a feature flag.